### PR TITLE
Fix links to PostgreSQL JDBC docs.

### PR DESCRIPTION
**Pull request by ianjevans**
The PostgreSQL JDBC docs were refactored, and our external links all broke. This PR fixes the old links with their new structure.

Fixes DOC-6386.
Fixes DOC-6478.

### CODE REVIEW

File: v19.1/build-a-java-app-with-cockroachdb-hibernate.md [hunk 1]
```diff
@@ -266,7 +266,7 @@ For more information about importing data from MySQL, see [Migrate from MySQL](m
 
 ### Use `rewriteBatchedInserts` for increased speed
 
-We strongly recommend setting `rewriteBatchedInserts=true`; we have seen 2-3x performance improvements with it enabled. From [the JDBC connection parameters documentation](https://jdbc.postgresql.org/documentation/head/connect.html#connection-parameters):
+We strongly recommend setting `rewriteBatchedInserts=true`; we have seen 2-3x performance improvements with it enabled. From [the JDBC connection parameters documentation](https://jdbc.postgresql.org/documentation/use/#connection-parameters):
```
Comment (thatnerd) [inline v19.1/build-a-java-app-with-cockroachdb-hibernate.md:269]: Oh good. I was looking at the old one and I don't think it was even using the fragment correctly before the link broke.

File: v21.2/connect-to-the-database.md [hunk 2]
```diff
@@ -931,7 +931,7 @@ Parameter | Description
 
 <div class="filter-content" markdown="1" data-scope="jdbc">
 
-To connect to CockroachDB with the [JDBC](https://jdbc.postgresql.org) driver, create a `DataSource` object ([`PGSimpleDataSource` or `PGPoolingDataSource`](https://jdbc.postgresql.org/documentation/head/ds-ds.html)), and set the connection string with the `setUrl` class method.
+To connect to CockroachDB with the [JDBC](https://jdbc.postgresql.org) driver, create a `DataSource` object ([`PGSimpleDataSource` or `PGPoolingDataSource`](hhttps://jdbc.postgresql.org/documentation/datasource/#applications-datasource)), and set the connection string with the `setUrl` class method.
```
Comment (thatnerd) [inline v21.2/connect-to-the-database.md:934]: Bug here: `hhtps` -> `https`

File: v22.1/connect-to-the-database.md [hunk 3]
```diff
@@ -922,7 +922,7 @@ Parameter | Description
 
 <div class="filter-content" markdown="1" data-scope="jdbc">
 
-To connect to CockroachDB with the [JDBC](https://jdbc.postgresql.org) driver, create a `DataSource` object ([`PGSimpleDataSource` or `PGPoolingDataSource`](https://jdbc.postgresql.org/documentation/head/ds-ds.html)), and set the connection string with the `setUrl` class method.
+To connect to CockroachDB with the [JDBC](https://jdbc.postgresql.org) driver, create a `DataSource` object ([`PGSimpleDataSource` or `PGPoolingDataSource`](hhttps://jdbc.postgresql.org/documentation/datasource/#applications-datasource)), and set the connection string with the `setUrl` class method.
```
Comment (thatnerd) [inline v22.1/connect-to-the-database.md:925]: `hhttps` -> `https`

File: v22.2/connect-to-the-database.md [hunk 4]
```diff
@@ -925,7 +925,7 @@ Parameter | Description
 
 <div class="filter-content" markdown="1" data-scope="jdbc">
 
-To connect to CockroachDB with the [JDBC](https://jdbc.postgresql.org) driver, create a `DataSource` object ([`PGSimpleDataSource` or `PGPoolingDataSource`](https://jdbc.postgresql.org/documentation/head/ds-ds.html)), and set the connection string with the `setUrl` class method.
+To connect to CockroachDB with the [JDBC](https://jdbc.postgresql.org) driver, create a `DataSource` object ([`PGSimpleDataSource` or `PGPoolingDataSource`](hhttps://jdbc.postgresql.org/documentation/datasource/#applications-datasource)), and set the connection string with the `setUrl` class method.
```
Comment (thatnerd) [inline v22.2/connect-to-the-database.md:928]: `hhttps` -> `https`

**Review by thatnerd — CHANGES_REQUESTED** [review]
Small (but repeated) nit: there are 9 instances of `hhttps` rather than `https` (I only caught a few visually but I got the rest with `grep -Iir 'hhttps' .` in the root directory of the repo.

**Review by nickvigilante — CHANGES_REQUESTED** [review]
Same suggestions as @thatnerd, but otherwise LGTM! I'm gonna send a PR to tidy up some of the false Vale errors.

**Review by nickvigilante — APPROVED** [review]

**Pull request merged**